src/presentation/ui/page/databases.templ

```diff
@@ -0,0 +1,352 @@
+				"Databases",
+				"Install the database services you need and manage each service individually by adding databases and users. Control user permissions by specifying what each user can and cannot do.",
+				"ph-database",
+			)
+		</div>
+		<div
+			id="databases-page-content"
+			hx-get={ "/databases/?dbType=" + selectedDatabaseTypeOverview.Type.String() }
+			hx-trigger="submit from:form delay:500ms, click from:button#delete-element delay:500ms"
+			hx-select="#databases-page-content"
+			hx-target="#databases-page-content"
+			hx-swap="outerHTML transition:true"
+			class="card w-full"
+		>
+			<div id="databases-tabs">
+				@DatabasesTabs(selectedDatabaseTypeOverview)
+			</div>
+			@CreateDatabaseModal(selectedDatabaseTypeOverview.Type)
+			@componentStructural.DeleteModal(
+				"isDeleteDatabaseModalOpen", "closeDeleteDatabaseModal()",
+				"deleteDatabaseElement('"+selectedDatabaseTypeOverview.Type.String()+"')",
+				"database.name", "",
+			)
+			@CreateDatabaseUserModal(selectedDatabaseTypeOverview)
+			@componentStructural.DeleteModal(
+				"isDeleteDatabaseUserModalOpen", "closeDeleteDatabaseUserModal()",
+				"deleteDatabaseUserElement('"+selectedDatabaseTypeOverview.Type.String()+"')",
+				"databaseUser.username", "",
+			)
+		</div>
+	</div>
+}
+
+func getDatabasesTabHeaderItems(
+	selectedDatabaseType valueObject.DatabaseType,
+) []componentStructural.TabHeaderItem {
+	tabHeaderItems := []componentStructural.TabHeaderItem{
+		{Label: "MySQL/MariaDB/Percona", ComponentValue: "mariadb"},
+		{Label: "PostgreSQL", ComponentValue: "postgresql"},
+	}
+
+	selectedDbTypeStr := selectedDatabaseType.String()
+	for itemIndex := range tabHeaderItems {
+		if selectedDbTypeStr != tabHeaderItems[itemIndex].ComponentValue {
+			tabHeaderItems[itemIndex].AdditionalClasses = "bg-os-700 hover:bg-os-300 text-slate-500"
+			continue
+		}
+
+		tabHeaderItems[itemIndex].AdditionalClasses = "bg-os-500 text-neutral-50"
+	}
+
+	return tabHeaderItems
+}
+
+templ DatabaseServiceNotInstalledWarning(selectedDbType valueObject.DatabaseType) {
+	<div class="px-4 pb-4 pt-2 bg-os-500 rounded-r-lg">
+		<i class="ph-duotone ph-warning align-middle text-xl text-yellow mr-3"></i>
+		<span class="mr-3">The<p class="inline-flex mx-1 text-speedia-300 font-bold">{ selectedDbType.String() }</p>service is not installed yet.</span>
+		@componentForm.SubmitButton(
+			"Schedule "+selectedDbType.String()+" service installation", "ph-caret-line-down",
+			"installDatabaseService('"+selectedDbType.String()+"')", true,
+		)
+	</div>
+}
+
+templ DatabasesTabs(selectedDatabaseTypeOverview presenterDto.DatabaseOverview) {
+	@componentStructural.TabHeader(
+		getDatabasesTabHeaderItems(selectedDatabaseTypeOverview.Type), "changeSelectedDatabaseType",
+	)
+	<!-- DatabaseTabContent -->
+	<div class="bg-os-800 p-4 -mt-4 rounded-b-xl rounded-r-xl">
+		<div class="px-4 bg-os-500 rounded-b-lg rounded-r-lg space-y-3 pt-2">
+			if selectedDatabaseTypeOverview.IsInstalled {
```

Comment:
Hm, I think I know now a way to avoid this if. Imagine if line 153 (back on DatabasesIndex) was like this:

```
<div id="databases-tabs">
	  @componentStructural.TabHeader(
		  getDatabasesTabHeaderItems(selectedDbOverview.Type), "changeSelectedDatabaseType",
	  )
	  if len(selectedDbOverview.Databases) > 0 {
					  @DatabasesTabsContent(selectedDbOverview)
	  } else {
					  @DatabaseServiceNotInstalledWarning(selectedDbOverview.Type)
	  }
</div>
```

Neat. What do you think? It's just a suggestion tho. That way your TabsContent is only used for when the service is installed.

Reply:
I still believe it makes more sense to keep it out of `DatabaseIndex()`, as that's where we build the entire page, separating the actual content from the page header. This way, we ensure that pages with `tabs` as content maintain a consistent index layout, since it's the content that changes, not the header.

Additionally, checking whether a database needs to be installed based on the number of existing databases (`len(selectedDbOverview.Databases) > 0`) wouldn’t work, as a database service could be installed, but no databases may have been created yet.